public/app/features/commandPalette/actions/useActions.tsx

```diff
@@ -0,0 +1,194 @@
+import { useRegisterActions } from 'kbar';
+import { ReactNode, useCallback, useEffect, useMemo, useState } from 'react';
+import { useObservable } from 'react-use';
+import { Observable } from 'rxjs';
+
+import { config } from '@grafana/runtime';
+
+import { t } from '../../../core/internationalization';
+import { useScopesServices } from '../../scopes/ScopesContextProvider';
+import { ScopesSelectorServiceState } from '../../scopes/selector/ScopesSelectorService';
+import { NodesMap, Node, TreeScope, ToggleNode } from '../../scopes/selector/types';
+import { ScopesRow } from '../ScopesRow';
+import { CommandPaletteAction } from '../types';
+import { SCOPES_PRIORITY } from '../values';
+
+import { getRecentDashboardActions } from './dashboardActions';
+import { getRecentScopesActions } from './recentScopesActions';
+import { useStaticActions } from './staticActions';
+import useExtensionActions from './useExtensionActions';
+
+/**
+ * Register navigation actions to different parts of grafana or some preferences stuff like themes.
+ */
+export function useRegisterStaticActions() {
```

Comment:
Main change is that each type of actions have their own hook and each hook also register the actions itself. This make it easier to decide when to load the actions (like before opening the cmdK or after) and also handle the dependency of the register call so action are not re-registered all the time if they are dynamic.